ambry-api/src/main/java/com.github.ambry/clustermap/VirtualReplicatorCluster.java

```diff
@@ -34,6 +35,12 @@
    */
   DataNodeId getCurrentDataNodeId();
 
+  /**
+   * Join the cluster as {@link InstanceType#PARTICIPANT} or {@link InstanceType#SPECTATOR} .
+   * @param role the {@link InstanceType}.
+   */
+  void participate(InstanceType role) throws Exception;
```

Comment:
Not good to expose Helix artifacts in an interface that is supposed to be generic.  Prefer to split this into two methods, participate() and spectate().
Also, it doesn't look like this is ever called with the SPECTATOR role.  Will it be?

Reply:
`SPECTATOR ` will be called for recoveryManager.
HelixClustrerMap somehow used the same method, so I followed it. Open to split it.

Reply:
If you mean ClusterParticipant, that participate method doesn't have the Helix enum argument and doesn't seem to have the spectator option.  If you want to keep it as one method, then I suggest adding our own enum that maps to Helix's.  But I prefer to split it.